File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/property/PropertyType.js
define([ "stjs", "util/Exceptions" ], function(stjs, exceptions) {
	var PropertyType = function() {
	};
	stjs.extend(PropertyType, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.property.PropertyType';
		constructor.serialVersionUID = 1;
		prototype.fetchOptions = null;
		prototype.code = null;
		prototype.permId = null;
		prototype.label = null;
		prototype.description = null;
		prototype.managedInternally = null;
		prototype.dataType = null;
		prototype.vocabulary = null;
		prototype.materialType = null;
		prototype.sampleType = null;
		prototype.schema = null;
		prototype.transformation = null;
		prototype.semanticAnnotations = null;
		prototype.registrator = null;
		prototype.registrationDate = null;
		prototype.metaData = null;

		prototype.getFetchOptions = function() {
			return this.fetchOptions;
		};
		prototype.setFetchOptions = function(fetchOptions) {
			this.fetchOptions = fetchOptions;
		};
		prototype.getCode = function() {
			return this.code;
		};
		prototype.setCode = function(code) {
			this.code = code;
		};
		prototype.getPermId = function() {
			return this.permId;
		};
		prototype.setPermId = function(permId) {
			this.permId = permId;
		};
		prototype.getLabel = function() {
			return this.label;
		};
		prototype.setLabel = function(label) {
			this.label = label;
		};
		prototype.getDescription = function() {
			return this.description;
		};
		prototype.setDescription = function(description) {
			this.description = description;
		};
		prototype.isManagedInternally = function() {
			return this.managedInternally;
		};
		prototype.setManagedInternally = function(managedInternally) {
			this.managedInternally = managedInternally;
		};
		prototype.isInternalNameSpace = function() {
			return this.isManagedInternally();
		};
		prototype.setInternalNameSpace = function(internalNameSpace) {
			this.setManagedInternally(internalNameSpace);
		};
		prototype.getDataType = function() {
			return this.dataType;
		};
		prototype.setDataType = function(dataType) {
			this.dataType = dataType;
		};
		prototype.getVocabulary = function() {
			if (this.getFetchOptions() && this.getFetchOptions().hasVocabulary()) {
				return this.vocabulary;
			} else {
				throw new exceptions.NotFetchedException("Vocabulary has not been fetched.");
			}
		};
		prototype.setVocabulary = function(vocabulary) {
			this.vocabulary = vocabulary;
		};
		prototype.getMaterialType = function() {
			if (this.getFetchOptions() && this.getFetchOptions().hasMaterialType()) {
				return this.materialType;
			} else {
				throw new exceptions.NotFetchedException("Material type has not been fetched.");
			}
		};
		prototype.setMaterialType = function(materialType) {
			this.materialType = materialType;
		};
		prototype.getSampleType = function() {
			if (this.getFetchOptions() && this.getFetchOptions().hasSampleType()) {
				return this.sampleType;
			} else {
				throw new exceptions.NotFetchedException("Sample type has not been fetched.");
			}
		};
		prototype.setSampleType = function(sampleType) {
			this.sampleType = sampleType;
		};
		prototype.getSchema = function() {
			return this.schema;
		};
		prototype.setSchema = function(schema) {
			this.schema = schema;
		};
		prototype.getTransformation = function() {
			return this.transformation;
		};
		prototype.setTransformation = function(transformation) {
			this.transformation = transformation;
		};
		prototype.getSemanticAnnotations = function() {
			if (this.getFetchOptions() && this.getFetchOptions().hasSemanticAnnotations()) {
				return this.semanticAnnotations;
			} else {
				throw new exceptions.NotFetchedException("Semantic annotations have not been fetched.");
			}
		};
		prototype.setSemanticAnnotations = function(semanticAnnotations) {
			this.semanticAnnotations = semanticAnnotations;
		};
		prototype.getRegistrator = function() {
			if (this.getFetchOptions() && this.getFetchOptions().hasRegistrator()) {
				return this.registrator;
			} else {
				throw new exceptions.NotFetchedException("Registrator has not been fetched.");
			}
		};
		prototype.setRegistrator = function(registrator) {
			this.registrator = registrator;
		};
		prototype.getRegistrationDate = function() {
			return this.registrationDate;
		};
		prototype.setRegistrationDate = function(registrationDate) {
			this.registrationDate = registrationDate;
		};
		prototype.getMetaData = function() {
			return this.metaData;
		};
		prototype.setMetaData = function(metaData) {
			this.metaData = metaData;
		};
	}, {
		fetchOptions : "PropertyTypeFetchOptions",
		permId : "PropertyTypePermId",
		dataType : "DataType",
		vocabulary : "Vocabulary",
		materialType : "MaterialType",
		sampleType : "SampleType",
		registrator : "Person",
		registrationDate : "Date"
	});
	return PropertyType;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/property/PropertyType.java
/*
 * Copyright 2014 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ch.ethz.sis.openbis.generic.asapi.v3.dto.property;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.ICodeHolder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.IDescriptionHolder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.IPermIdHolder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.IRegistrationDateHolder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.IRegistratorHolder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.material.MaterialType;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.person.Person;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.DataType;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.fetchoptions.PropertyTypeFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.id.PropertyTypePermId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.vocabulary.Vocabulary;
import ch.ethz.sis.openbis.generic.asapi.v3.exceptions.NotFetchedException;
import ch.systemsx.cisd.base.annotation.JsonObject;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.io.Serializable;
import java.util.Date;

/*
 * Class automatically generated with DtoGenerator
 */
@JsonObject("as.dto.property.PropertyType")
public class PropertyType implements Serializable, ICodeHolder, IDescriptionHolder, IPermIdHolder, IRegistrationDateHolder, IRegistratorHolder
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private PropertyTypeFetchOptions fetchOptions;

    @JsonProperty
    private String code;

    @JsonProperty
    private PropertyTypePermId permId;

    @JsonProperty
    private String label;

    @JsonProperty
    private String description;

    @JsonProperty
    private Boolean managedInternally;

    @JsonProperty
    private Boolean internalNameSpace;

    @JsonProperty
    private DataType dataType;

    @JsonProperty
    private Vocabulary vocabulary;

    @JsonProperty
    private MaterialType materialType;

    @JsonProperty
    private String schema;

    @JsonProperty
    private String transformation;

    @JsonProperty
    private Person registrator;

    @JsonProperty
    private Date registrationDate;

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public PropertyTypeFetchOptions getFetchOptions()
    {
        return fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public void setFetchOptions(PropertyTypeFetchOptions fetchOptions)
    {
        this.fetchOptions = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    @Override
    public String getCode()
    {
        return code;
    }

    // Method automatically generated with DtoGenerator
    public void setCode(String code)
    {
        this.code = code;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    @Override
    public PropertyTypePermId getPermId()
    {
        return permId;
    }

    // Method automatically generated with DtoGenerator
    public void setPermId(PropertyTypePermId permId)
    {
        this.permId = permId;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public String getLabel()
    {
        return label;
    }

    // Method automatically generated with DtoGenerator
    public void setLabel(String label)
    {
        this.label = label;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    @Override
    public String getDescription()
    {
        return description;
    }

    // Method automatically generated with DtoGenerator
    public void setDescription(String description)
    {
        this.description = description;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public Boolean isManagedInternally()
    {
        return managedInternally;
    }

    // Method automatically generated with DtoGenerator
    public void setManagedInternally(Boolean managedInternally)
    {
        this.managedInternally = managedInternally;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public Boolean isInternalNameSpace()
    {
        return internalNameSpace;
    }

    // Method automatically generated with DtoGenerator
    public void setInternalNameSpace(Boolean internalNameSpace)
    {
        this.internalNameSpace = internalNameSpace;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public DataType getDataType()
    {
        return dataType;
    }

    // Method automatically generated with DtoGenerator
    public void setDataType(DataType dataType)
    {
        this.dataType = dataType;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public Vocabulary getVocabulary()
    {
        if (getFetchOptions() != null && getFetchOptions().hasVocabulary())
        {
            return vocabulary;
        }
        else
        {
            throw new NotFetchedException("Vocabulary has not been fetched.");
        }
    }

    // Method automatically generated with DtoGenerator
    public void setVocabulary(Vocabulary vocabulary)
    {
        this.vocabulary = vocabulary;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public MaterialType getMaterialType()
    {
        if (getFetchOptions() != null && getFetchOptions().hasMaterialType())
        {
            return materialType;
        }
        else
        {
            throw new NotFetchedException("Material type has not been fetched.");
        }
    }

    // Method automatically generated with DtoGenerator
    public void setMaterialType(MaterialType materialType)
    {
        this.materialType = materialType;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public String getSchema()
    {
        return schema;
    }

    // Method automatically generated with DtoGenerator
    public void setSchema(String schema)
    {
        this.schema = schema;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public String getTransformation()
    {
        return transformation;
    }

    // Method automatically generated with DtoGenerator
    public void setTransformation(String transformation)
    {
        this.transformation = transformation;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    @Override
    public Person getRegistrator()
    {
        if (getFetchOptions() != null && getFetchOptions().hasRegistrator())
        {
            return registrator;
        }
        else
        {
            throw new NotFetchedException("Registrator has not been fetched.");
        }
    }

    // Method automatically generated with DtoGenerator
    public void setRegistrator(Person registrator)
    {
        this.registrator = registrator;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    @Override
    public Date getRegistrationDate()
    {
        return registrationDate;
    }

    // Method automatically generated with DtoGenerator
    public void setRegistrationDate(Date registrationDate)
    {
        this.registrationDate = registrationDate;
    }

    // Method automatically generated with DtoGenerator
    @Override
    public String toString()
    {
        return "PropertyType " + code;
    }

}
